### PR TITLE
Tag Servlet

### PR DESCRIPTION
**Pull request by hari387**
Adds the following:
- `getTagsMatchingIds` function in `DBInterface` and `FirestoreDB`
- `TagServlet` which responds to different GET requests for tags
- Tests for said `TagServlet` in `TagServletTest`

### CODE REVIEW

File: src/main/java/com/google/sps/data/Tag.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 public class Tag extends DBObject {
   public static final String HIDDEN_KEY = "hidden";
+  public static final String ID_KEY = "id";
```
Comment (bradleypickard) [inline src/main/java/com/google/sps/data/Tag.java:5]: Could this property be migrated to DBObject, since it's common to all DBObjects?

File: src/main/java/com/google/sps/data/Tag.java [hunk 2]
```diff
@@ -14,4 +15,15 @@ public Tag() {
...
+  
+  public Tag(String id, String name, boolean hidden, String type) {
+    super(id);
+    this.name = name;
+    this.hidden = hidden;
+    this.type = type;
+  }
+
+  public String toString() {
+    return "Tag[id=" + super.id + ", name=" + name + ", hidden=" + String.valueOf(hidden) + ", type=" + type + "}";
```
Comment (hanpeiz) [inline src/main/java/com/google/sps/data/Tag.java:26]: what is this being used for? Also it is starting with [ and ending with }.

File: src/main/java/com/google/sps/servlets/TagServlet.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.meltingpot.servlets;
+
+import com.google.api.core.ApiFuture;
```
Comment (hanpeiz) [inline src/main/java/com/google/sps/servlets/TagServlet.java:17]: Do you have an import cleaner tool/command that removes unused imports?

File: src/main/java/com/google/sps/data/FirestoreDB.java [hunk 4]
```diff
@@ -102,26 +102,34 @@ public Long voteRecipe(String Id, int voteDiff) {
...
+    return DBUtils.blockOnFuture(tagsQuery.get()).toObjects(Tag.class);
+  }
   
   public boolean isDocument(String docId, String collection) {
     DocumentReference docRef = DBUtils.database.collection(collection).document(docId);
     DocumentSnapshot document = DBUtils.blockOnFuture(docRef.get());
     return document.exists();
   }
 
+  /** Returns a User object from userId. */
```
Comment (hanpeiz) [inline src/main/java/com/google/sps/data/FirestoreDB.java:117]: Why did you add these comments?

File: src/main/java/com/google/sps/servlets/TagServlet.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String[] tagIds = request.getParameterValues("tagIds");
+    String getHidden = request.getParameter("getHidden");
+
+    List<Tag> tags;
+
+    if (tagIds == null) {
```
Comment (hanpeiz) [inline src/main/java/com/google/sps/servlets/TagServlet.java:73]: what happens if tagIds exist but has no value? like:
https://our.url.com/Tagservlet?tagIds=&getHidden=true?

File: src/main/java/com/google/sps/data/Tag.java (hunk 2 above)
Comment (hari387) [reply src/main/java/com/google/sps/data/Tag.java:26]: I thought it might be useful for debugging (`toString` is used in all the printing functions)

File: src/main/java/com/google/sps/servlets/TagServlet.java (hunk 3 above)
Comment (hari387) [reply src/main/java/com/google/sps/servlets/TagServlet.java:17]: Not yet, I probably should get one and clean this up.

File: src/main/java/com/google/sps/servlets/TagServlet.java (hunk 5 above)
Comment (hari387) [reply src/main/java/com/google/sps/servlets/TagServlet.java:73]: That would count as the empty string, and since no tag has the empty string for an id, it would simply return an empty array.

File: src/main/java/com/google/sps/data/FirestoreDB.java (hunk 4 above)
Comment (hari387) [reply src/main/java/com/google/sps/data/FirestoreDB.java:117]: I think they are from when I merged master into my branch. Ill delete them.
